thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftSerializationFormats.java

```diff
@@ -55,14 +63,44 @@
      */
     public static final SerializationFormat TEXT_NAMED_ENUM = SerializationFormat.of("ttext-named-enum");
 
-    private static final Set<SerializationFormat> THRIFT_FORMATS =
-            ImmutableSet.of(BINARY, COMPACT, JSON, TEXT, TEXT_NAMED_ENUM);
+    /**
+     * A way to lookup the related {@link TProtocolFactory} from a {@link SerializationFormat}.
+     * Entries are provided via registered SPI {@link ThriftProtocolFactoryProvider} implementations.
+     */
+    private static final Map<SerializationFormat, TProtocolFactory> knownProtocolFactories;
+
+    static {
+        final List<ThriftProtocolFactoryProvider> providers = ImmutableList.copyOf(
+                ServiceLoader.load(ThriftProtocolFactoryProvider.class,
+                                   ThriftProtocolFactoryProvider.class.getClassLoader()));
+        knownProtocolFactories = providers
+                .stream()
+                .map(ThriftProtocolFactoryProvider::entries)
+                .flatMap(Set::stream)
+                .collect(toImmutableMap(Entry::getSerializationFormat,
+                                        Entry::getTProtocolFactory));
+    }
+
+    /**
+     * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
+     *
+     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not a
+     *         known Thrift serialization format
+     */
+    public static TProtocolFactory get(SerializationFormat serializationFormat) {
+        requireNonNull(serializationFormat, "serializationFormat");
+        return Optional.ofNullable(knownProtocolFactories.get(serializationFormat))
+                       .orElseThrow(() -> new IllegalArgumentException(
+                               "Unsupported Thrift serializationFormat: " + serializationFormat));
```

Comment:
nit: null checking with if condition? The `Optional` seems to only be in a stack by escape analysis, however, we always try to avoid creating additional objects if possible.

Reply:
I suppose it's mostly a matter of taste. changed.